packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -32,32 +35,56 @@ class WeatherScreen extends HookWidget {
 
     final fullWeatherStateNotifier =
         useProvider(w.fullWeatherStateNotifierProvider.notifier);
+
     final controller = useFloatingSearchBarController();
 
-    final isLoading = useState(fullWeatherState is c.Loading);
+    final isCityChanging = useState(false);
 
     final cityStateNotifier = useProvider(c.cityStateNotifierProvider.notifier);
 
-    Future<void> load() async {
-      await fullWeatherStateNotifier.loadFullWeather();
-    }
+    final refreshIndicatorKey = useGlobalKey<RefreshIndicatorState>();
 
     useEffect(
-      () => cityStateNotifier.addListener((state) {
-        if (state is c.Error) {
-          showFailureSnackBar(context, failure: state.failure, duration: 2);
+      () {
+        Future<void> load() async {
+          await Future.microtask(() async {
+            await fullWeatherStateNotifier.loadFullWeather();
+          });
```

Comment:
This was only made a function before because we had to call it recursively. We can just create the future now directly, and then remove the `load()` line below (I would've done that in this suggestion if GitHub didn't let me, apparently because there are deleted lines between this part and `load()`):
```suggestion
        Future.microtask(() => fullWeatherStateNotifier.loadFullWeather());
```

(Changing the block function to an arrow function is an unrelated simplification.)